workers/unity/Assets/Playground/Scripts/Worker/ConnectGameLogicWorker.cs

```diff
@@ -7,74 +7,85 @@
     public int ConnectionAttempts = 3;
 
     public Worker GameLogicWorker;
 
-    protected void Start()
+    protected async void Start()
     {
-        CreateWorker(ConnectionAttempts);
+        await CreateWorkerAsync(ConnectionAttempts);
     }
 
     protected void OnApplicationQuit()
     {
-        GameLogicWorker?.Dispose();
+        Dispose();
     }
 
-    private void CreateWorker(int attempts)
+    private async Task CreateWorkerAsync(int attempts)
     {
-        Task<Worker> workerTask;
         const string workerType = SystemConfig.UnityGameLogic;
+        ReceptionistConfig config;
         if (Application.isEditor)
         {
-            var config = new ReceptionistConfig
+            config = new ReceptionistConfig
             {
                 WorkerType = workerType,
                 WorkerId = $"{workerType}-{Guid.NewGuid()}"
             };
-            workerTask = Worker.CreateWorkerAsync(config, new ForwardingDispatcher(), Origin);
         }
         else
         {
             var commandLineArguments = Environment.GetCommandLineArgs();
-            Debug.LogFormat("Command line {0}", string.Join(" ", commandLineArguments.ToArray()));
             var commandLineArgs = CommandLineUtility.ParseCommandLineArgs(commandLineArguments);
-            var config = ReceptionistConfig.CreateConnectionConfigFromCommandLine(commandLineArgs);
+            config = ReceptionistConfig.CreateConnectionConfigFromCommandLine(commandLineArgs);
             if (!commandLineArgs.ContainsKey(RuntimeConfigNames.WorkerId))
             {
                 config.WorkerId = $"{workerType}-{Guid.NewGuid()}";
             }
-
-            workerTask = Worker.CreateWorkerAsync(config, new ForwardingDispatcher(), Origin);
         }
 
-        StartCoroutine(CheckForWorkerReady(workerTask, attempts - 1));
-    }
-
-    private IEnumerator CheckForWorkerReady(Task<Worker> workerTask, int attempts)
-    {
-        while (!workerTask.IsCompleted)
+        try
         {
-            yield return null;
+            var worker = await ConnectWithRetriesAsync(config, new ForwardingDispatcher(), Origin, attempts);
+            InitializeWorker(worker);
         }
+        catch (Exception e)
+        {
+            Debug.LogError($"Failed to create worker {config.WorkerId} with message:{Environment.NewLine}{e.Message}");
+            Dispose();
+        }
+    }
 
-        if (workerTask.IsFaulted)
+    private async Task<Worker> ConnectWithRetriesAsync(ReceptionistConfig config, ILogDispatcher dispatcher, Vector3 origin,
+        int attempts)
+    {
+        while (attempts > 0)
         {
-            Debug.LogError(
-                $"Failed to create game logic worker with error: {workerTask.Exception.InnerException.Message}");
-            if (attempts > 0)
+            try
             {
-                CreateWorker(attempts);
+                var worker = await Worker.CreateWorkerAsync(config, dispatcher, origin);
```

Comment:
Minor thing, I was confused by CreateWorkerAsync as I thought for a second that it was the method at the top of the file.
Tbh that's on me, but if we wanted to be extra clear, the private method in this file could be
'private ConnectGameLogicWorkerAsync" showing it's the specialised version.